project/Build.scala

```diff
@@ -501,7 +504,7 @@ object Build {
       // For convenience, change the baseDirectory when running the compiler
       baseDirectory in (Compile, run) := baseDirectory.value / "..",
       // And when running the tests
-      baseDirectory in Test := baseDirectory.value / "..",
+      baseDirectory in (Test, run) := baseDirectory.value / "..",
```

Comment:
I could use some help here... The reason for this change it breaks the test directory in the bsp output.

Before the change:

```
sbt:scala3> show scala3-compiler/Test/bspWorkspaceBuildTargets
...
[info] * BuildTarget(
  BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-compiler/Compile),
  Some(scala3-compiler),
  Some(file:/Users/luc/scala/dotty/compiler/),
  Vector(library), BuildTargetCapabilities(true, true, true), Vector(scala), Vector(BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-library/Compile), BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-interfaces/Compile), BuildTargetIdentifier(file:/Users/luc/scala/dotty/#tasty-core/Compile)), Some(scala), Some(JObject([Lsjsonnew.shaded.scalajson.ast.unsafe.JField;@2d404c74)))

[info] * BuildTarget(
  BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-compiler/Test),
  Some(scala3-compiler-test),
  Some(file:/Users/luc/scala/dotty/compiler/../),
  Vector(test), BuildTargetCapabilities(true, true, true), Vector(scala), Vector(BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-compiler/Compile), BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-library/Compile), BuildTargetIdentifier(file:/Users/luc/scala/dotty/#scala3-interfaces/Compile), BuildTargetIdentifier(file:/Users/luc/scala/dotty/#tasty-core/Compile)), Some(scala), Some(JObject([Lsjsonnew.shaded.scalajson.ast.unsafe.JField;@d6e4af0)))
```

The `../` in `scala3-compiler-test` should not be there, it breaks the IntelliJ import.

My change fixes the bsp output, but it breaks running the tests.

After the change:

```
sbt:scala3> scala3-compiler/testOnly dotty.tools.dotc.core.tasty.PathPicklingTest*
[error] Test dotty.tools.dotc.core.tasty.PathPicklingTest.test failed: java.nio.file.NoSuchFileException: /Users/luc/scala/dotty/compiler/out/testPathPickling/out.jar, took 0.709 sec
```

Anyone knows how to scope this? I tried various things. IIUC, `scala3-compiler/Test/baseDirectory` has to stay unchanged for the [`bspWorkspaceBuildTargets` implementation to work](https://github.com/sbt/sbt/blob/0972b7fa174ab4dd3d6332835e7f01a9370c8524/main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala#L97-L106). Is there a way to change the `baseDirectory` when running the tests?

Reply:
Just saw that the [sbt docutmentation suggests to set `baseDirectory in Test`](https://www.scala-sbt.org/1.x/docs/Forking.html#Change+working+directory)... cc @adpi2 that this affects `bspWorkspaceBuildTargets`.

Reply:
We already found this fix in the scala/scala build:

```scala
(Test / forkOptions) := (Test / forkOptions).value.withWorkingDirectory((ThisBuild / baseDirectory).value),
(Test / testOnly / forkOptions) := (Test / testOnly / forkOptions).value.withWorkingDirectory((ThisBuild / baseDirectory).value),
```

Reply:
Thanks @lrytz for pointing this out to me. At first sight I would think it's a limitation of IntelliJ since the BSP documentation clearly states that 2 build targets can have the same `baseDirectory`:
```
  /** The directory where this target belongs to. Multiple build targets are allowed to map
   * to the same base directory, and a build target is not required to have a base directory.
   * A base directory does not determine the sources of a target, see buildTarget/sources. */
  baseDirectory?: Uri;
```
I have created https://youtrack.jetbrains.com/issue/SCL-18759 to track this.

But, at the same time it is unfortunate that changing the working directory, using the recommended method, affects the definition of the build target: https://github.com/sbt/sbt/issues/6345

Reply:
Thank you for creating these follow-up tickets!